### PR TITLE
Feature/new route

### PR DESCRIPTION
**Pull request by TawfiqKhan**
### What this PR does (required):
- route for creating board, getting an existing board
- create column

### Screenshots / Videos (required):
-N/A

### Any information needed to test this feature (required):
- pass in the required data in the post call

### Any issues with the current functionality (optional):
- route for cards, updating board/ column has not been finished yet

### CODE REVIEW

File: server/controllers/board.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+// Create new board
+exports.createBoard = asyncHandler(async (req, res, next)=> {
+  
+  const user = await User.findById(req.user.id);
+  if (!user) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+  const {title} = req.body
```
Comment (ajaytomgeorge) [inline server/controllers/board.js:16]: What happens if the req,body is empty? // Checking for empty input

File: server/controllers/card.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  const card = await Card.findById(req.params.id)
+  card.name = name
+  card.tag  = tag
+  card.save()
+  res.status(200).json(card)
+})
+
+// Move Card between columns
+exports.moveCard = asyncHandler(async (req, res)=> {
+  const {currentColumnId, newColumnId, sourceIndex, destinationIndex} = req.body
```
Comment (ajaytomgeorge) [inline server/controllers/card.js:37]: here too

File: server/controllers/card.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  if(currentColumnId === newColumnId) {
+    currentColumn.cards.splice(sourceIndex, 1)
+    currentColumn.cards.splice(destinationIndex, 0, card._id)
+    card.save()
+  }
+
+  // if the card moved tod a different column, first update card's column, remove it from current column, 
+  // add it in the new column
+
+  if(currentColumnId !== newColumnId) {
```
Comment (ajaytomgeorge) [inline server/controllers/card.js:51]: you can use if-else

**Review by ajaytomgeorge — COMMENTED** [review]
@TawfiqKhan Looks pretty good . Cheers :-) Some minor comments though

File: server/controllers/board.js (hunk 1 above)
Comment (TawfiqKhan) [reply server/controllers/board.js:16]: The front end will have validations. But I have added some error handling in the backend.

File: server/controllers/card.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+
+exports.createCard = asyncHandler(async (req, res, next)=> {
+
+  if (!req.body) {
+    res.status(204);
+    throw new Error("Input missing. please fill up all the fields.");
+  }
+
+  const {columnId, boardId, name, tag} = req.body
```
Comment (moffatethan) [inline server/controllers/card.js:14]: What if there is partial data passed down like no boardId? Ideally we want to do input validation on the back-end with middleware like express-validator.

File: server/controllers/column.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+const User = require("../models/User");
+const Board = require("../models/Board");
+const Column = require("../models/Column");
+const Card = require("../models/Card");
+const asyncHandler = require("express-async-handler");
+
+
+exports.createColumn = asyncHandler(async (req, res)=> {
+
+  const {boardId, title} = req.body
```
Comment (moffatethan) [inline server/controllers/column.js:10]: Same here, some input validation could be beneficial

Comment (moffatethan) [reply server/controllers/column.js:10]: On the title property

**Review by moffatethan — COMMENTED** [review]
Left some comments for you to address. Thanks!

File: server/controllers/card.js (hunk 4 above)
Comment (TawfiqKhan) [reply server/controllers/card.js]: Added validation using express-validator.

**Review by moffatethan — APPROVED** [review]
Good improvements 😄

**Pull request merged**